lib/matplotlib/axes/_axes.py

```diff
@@ -1076,8 +1076,8 @@ def axvspan(self, xmin, xmax, ymin=0, ymax=1, **kwargs):
 
     @_preprocess_data(replace_names=["y", "xmin", "xmax", "colors"],
                       label_namer="y")
-    def hlines(self, y, xmin, xmax, colors='k', linestyles='solid',
-               label='', **kwargs):
+    def hlines(self, y, xmin, xmax, colors=rcParams['lines.color'] or 'k',
```

Comment:
This needs to default to `None` and then in the body of the function check do this check.

The default values of a function are evaluated at function definition time so this will not reflect any changes to the rcparams.

Reply:
I'm assuming we have to make this change in all 4 functions that are affected within this PR?

Reply:
Yes, but the ones in pyplot should be done by running `boilerplate.py`